xtable-core/src/main/java/org/apache/xtable/catalog/glue/table/IcebergGlueCatalogTableBuilder.java

```diff
@@ -0,0 +1,121 @@
+
+/** Iceberg specific table operations for Glue catalog sync */
+public class IcebergGlueCatalogTableBuilder implements CatalogTableBuilder<TableInput, Table> {
+
+  private final GlueSchemaExtractor schemaExtractor;
+  private final HadoopTables hadoopTables;
+  private static final String tableFormat = TableFormat.ICEBERG;
+
+  public IcebergGlueCatalogTableBuilder(Configuration configuration) {
+    this.schemaExtractor = GlueSchemaExtractor.getInstance();
+    this.hadoopTables = new HadoopTables(configuration);
+  }
+
+  @VisibleForTesting
+  IcebergGlueCatalogTableBuilder(GlueSchemaExtractor schemaExtractor, HadoopTables hadoopTables) {
+    this.schemaExtractor = schemaExtractor;
+    this.hadoopTables = hadoopTables;
+  }
+
+  @Override
+  public TableInput getCreateTableRequest(
+      InternalTable table, CatalogTableIdentifier tableIdentifier) {
+    HierarchicalTableIdentifier tblIdentifier = castToHierarchicalTableIdentifier(tableIdentifier);
+    BaseTable fsTable = loadTableFromFs(table.getBasePath());
+    return TableInput.builder()
+        .name(tblIdentifier.getTableName())
+        .tableType(GLUE_EXTERNAL_TABLE_TYPE)
+        .parameters(getTableParameters(fsTable))
+        .storageDescriptor(
+            StorageDescriptor.builder()
+                .location(table.getBasePath())
+                .columns(schemaExtractor.toColumns(tableFormat, table.getReadSchema()))
+                .build())
+        .build();
+  }
+
+  @Override
+  public TableInput getUpdateTableRequest(
+      InternalTable table, Table catalogTable, CatalogTableIdentifier tableIdentifier) {
+    HierarchicalTableIdentifier tblIdentifier = castToHierarchicalTableIdentifier(tableIdentifier);
+    BaseTable icebergTable = loadTableFromFs(table.getBasePath());
+    Map<String, String> parameters = new HashMap<>(catalogTable.parameters());
+    parameters.put(PREVIOUS_METADATA_LOCATION_PROP, parameters.get(METADATA_LOCATION_PROP));
+    parameters.put(METADATA_LOCATION_PROP, getMetadataFileLocation(icebergTable));
+    parameters.putAll(icebergTable.properties());
+
+    return TableInput.builder()
+        .name(tblIdentifier.getTableName())
+        .tableType(GLUE_EXTERNAL_TABLE_TYPE)
+        .parameters(parameters)
+        .storageDescriptor(
+            StorageDescriptor.builder()
+                .location(table.getBasePath())
+                .columns(
+                    schemaExtractor.toColumns(tableFormat, table.getReadSchema(), catalogTable))
+                .build())
+        .build();
+  }
+
+  @VisibleForTesting
+  Map<String, String> getTableParameters(BaseTable icebergTable) {
+    Map<String, String> parameters = new HashMap<>(icebergTable.properties());
+    parameters.put(TABLE_TYPE_PROP, tableFormat);
+    parameters.put(METADATA_LOCATION_PROP, getMetadataFileLocation(icebergTable));
+    return parameters;
+  }
+
+  private BaseTable loadTableFromFs(String tableBasePath) {
```

Comment:
Will this work with an external catalog for Iceberg?

Reply:
what do you mean by "external catalog for iceberg" in this context?

Reply:
Production tables for Iceberg are managed with a catalog that handles the transactions, especially when dealing with multiple writer cases. This is why we allow users to specify that configuration when using Iceberg as a source or target for conversion today. In the conversion, we use that catalog to get the state.